Platform/CoffeelakeBoardPkg/Script/StitchLoader.py

```diff
@@ -235,17 +229,36 @@ def LocateComponent(Root, Path):
     else:
         return None
 
-def ReplaceRegion (IfwiData, Rgn, InputFile, TailPos, PlatformData = None):
-    RgnPos = Rgn[1]
-    RgnLen = Rgn[2]
-
+def AddPlatformData (InputFile, RgnLen, PlatformData = None):
     Fd = open(InputFile, "rb")
     InputData = bytearray(Fd.read())
     Fd.close()
 
     InputLen = len(InputData)
-    if RgnLen < InputLen:
-        Fatal("Input binary size(0x%08x) cannot exceed %s region size(0x%08x) !" % (InputLen, Rgn[0], RgnLen))
+    if RgnLen is not 0:
+        if RgnLen < InputLen:
+            Fatal("Input binary size(0x%08x) cannot exceed %s region size(0x%08x) !" % (InputLen, Rgn[0], RgnLen))
+            return None
+
+    if PlatformData:
+      # Parse Flash Map
+      Ifwi = ParseFlashMap (InputData)
+      for Part in range(2):
+        Path = 'IFWI/BIOS/TS%d/SG1A' %Part
+        Stage1A = LocateComponent (Ifwi, Path)
+        if Stage1A:
+          PlatDataOffset = Stage1A.Offset + Stage1A.Length - 12
+          c_uint32.from_buffer (InputData, PlatDataOffset).value = PlatformData
+        print "Platform data was patched for %s" % Path
+      return InputData
+
+def ReplaceRegion (IfwiData, Rgn, InputFile, TailPos, PlatformData = None):
+    RgnPos = Rgn[1]
+    RgnLen = Rgn[2]
+
+    InputData = AddPlatformData(InputFile, RgnLen, PlatformData)
```

Comment:
you need to add InputLen = len(InputData)

Reply:
Done